lib/collection/src/shard/mod.rs

```diff
@@ -24,27 +26,31 @@ pub type ShardId = u32;
 pub enum Shard {
     Local(LocalShard),
     Remote(RemoteShard),
+    Proxy(ProxyShard),
```

Comment:
I understand the naming is based on `ProxySegment` but I think in this case it doesn't capture the reason for this type of shard. In the end we would also need to encode the transferring shard state - so I would also rather do it now.

I suggest to have this structure
LocalShard:
- Normal
- Transferring:
    - From (and this would be our current ProxyShard)
    - To (rejects any updates, searches, etc. - only receives data during transfer)

Reply:
Proxy shard might be useful not only for transferring. I think transfer state should be external